### PR TITLE
Initial implementation of dashboard

### PR DESCRIPTION
**Pull request by oliverchang**
This PR includes a HTTP server to serve the dashboard.

A frontend using vue.js is also implemented.

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/firestore"
```
Comment (oliverchang) [inline server/main.go:8]: Using the firestore client library directly here as gocloud's docstore package had several limitations:

- No querying by "array-contains".
- No cursor support.

File: server/main.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+func queryHandler(c *gin.Context) {
+	var req Request
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	if req.Type != "file" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "invalid search type"})
```
Comment (laurentsimon) [inline server/main.go:60]: need to return?

Comment (oliverchang) [reply server/main.go:60]: good catch. fixed thanks!

**Comment by oliverchang** [comment]
I also added an implementation of the frontend to this PR. 

See example: https://ossf-malware-analysis.storage.googleapis.com/index.html

and https://ossf-malware-analysis.storage.googleapis.com/index.html#/package?ecosystem=pypi&name=mst-ssh&version=0.1.2

**Review by laurentsimon — APPROVED** [review]

File: web/dashboard/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  "name": "dashboard",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "serve": "vue-cli-service serve",
+    "build": "vue-cli-service build",
+    "lint": "vue-cli-service lint"
+  },
+  "dependencies": {
+    "bootstrap": "^5.0.1",
```
Comment (laurentsimon) [inline web/dashboard/package.json:11]: can these be pinned by hash? Maybe via package-lock.json?

File: web/dashboard/src/components/Search.vue [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+<script>
+export default {
+  name: 'Search',
+  props: {},
+  methods: {
+    async makeRequest() {
+      const curId = this.requestId;
+      const backend = process.env.VUE_APP_BACKEND;
+      const response = await fetch(
+          `${backend}/query`, {
```
Comment (laurentsimon) [inline web/dashboard/src/components/Search.vue:38]: do we need POST for a query, i.e.e read-only request? Can we use GET?

File: web/dashboard/src/views/Package.vue [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+<script>
+export default {
+  name: 'Package',
+  components: {},
+  methods: {
+    async makeRequest() {
+      const resultsUrl = process.env.VUE_APP_ANALYSIS_RESULTS;
+      const response = await fetch(
+          resultsUrl + '/' +
+          this.$route.query.ecosystem + '/' +
```
Comment (laurentsimon) [inline web/dashboard/src/views/Package.vue:25]: if there's a package to construct urls, we could use it.

Comment (oliverchang) [reply web/dashboard/src/views/Package.vue:25]: Don't think there's one builtin to Javascript..  Can revisit if this comes up more often.

File: web/dashboard/src/components/Search.vue (hunk 4 above)
Comment (oliverchang) [reply web/dashboard/src/components/Search.vue:38]: POST is also commonly used for query-like operations (e.g. https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query).

Also query parameters might become more complex over time, and it'll become unwieldy to keep using GET.

**Pull request merged**